webapp/lib/view.dart

```diff
@@ -1371,6 +1306,81 @@ class ProjectConfigurationView extends BaseView{
   }
 }
 
+abstract class BaseElement {
+  Element get renderElement;
+}
+
+class DropdownElement extends BaseElement {
+  Element _dropdown;
+
+  DropdownElement(List<String> dropdownItems, Function onItemSelected) {
```

Comment:
I suggest we change this constructor to take a map of dropdown items and their associated callback. That way, we don't need to store any info about the item in its dataset, and we don't need to have a switch inside the callback to figure out which item has been clicked.

Reply:
Refactored to a Map with callback